packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,276 @@
+            leftX(0.0), topY(0.86840689),
+            leftX(0.06549600), topY(0.66993427))
+        ..lineTo(leftX(0.07491100), topY(0.63149399))
+        ..cubicTo(leftX(0.16905899), topY(0.37282392),
+            leftX(0.37282401), topY(0.16906013),
+            leftX(0.63149399), topY(0.07491176))
+        ..cubicTo(leftX(0.86840701), topY(0),
+            leftX(1.08849299), topY(0.0),
+            leftX(1.52866483), topY(0.0))
+        ..close();
+    }
+
+    // The ratio of the declared corner radius to the total affected pixels to
+    // render the corner. For example if the declared radius were 25px then
+    // totalAffectedCornerPixelRatio * 25 (~38) pixels would be affected.
+    const double totalAffectedCornerPixelRatio = 1.52865;
+
+    // The radius multiplier where the resulting shape will concave with a
+    // height and width of any value.
+    //
+    // If the shortest side length to radius ratio drops below this value, the
+    // radius must be lessened to avoid clipping (ie. concavity) of the shape.
+    //
+    // This value comes from the website where the other equations and curves
+    // were found
+    // (https://www.paintcodeapp.com/news/code-for-ios-7-rounded-rectangles),
+    // however it represents the ratio of the total 90º curve width or height to
+    // the width or height of the smallest rectangle dimension.
+    const double minimalUnclippedSideToCornerRadiusRatio = 2.0 * totalAffectedCornerPixelRatio;
+
+    // The multiplier of the radius in comparison to the smallest edge length
+    // used to describe the minimum radius for this shape.
+    //
+    // This is multiplier used in the case of an extreme aspect ratio and a
+    // small extent value. It can be less than 'maxMultiplier' because there
+    // are not enough pixels to render the clipping of the shape at this size so
+    // it appears to still be concave (whereas mathematically it's convex).
+    const double minimalEdgeLengthSideToCornerRadiusRatio = 2.0;
+
+    // The minimum edge length at which the corner radius multiplier must be at
+    // its maximum so as to maintain the appearance of a perfectly concave,
+    // non-lozenge shape.
+    //
+    // If the smallest edge length is less than this value, the dynamic radius
+    // value can be made smaller than the 'maxMultiplier' while the rendered
+    // shape still does not visually clip.
+    const double minRadiusEdgeLength = 200.0;
+
+    final double minSideLength = math.min(rect.width, rect.height);
+
+    // As the minimum side edge length (where the round is occurring)
+    // approaches 0, the limitedRadius approaches 2 so as to maximize
+    // roundness (to make the shape with the largest radius that doesn't clip).
+    // As the edge length approaches 200, the limitedRadius approaches ~3 –- the
+    // multiplier of the radius value where the resulting shape is concave (ie.
+    // does not visually clip) at any dimension.
+    final double multiplier = ui.lerpDouble(
```

Comment:
Why is this needed? Do we think apple is doing this?

Reply:
This is to make a more rounded, stadium-esque shape at a small width or height.

Reply:
Same here, then this in-between-superellipse-and-stadium-ness should be in cupertino.